accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -1179,6 +1179,119 @@ all members of `V1, .. Vk` that are not redeclared by a declaration in _DV_
 can be invoked on a receiver of the type introduced by _DV_.*
 
 
+### Changes to type functions and types
+
+The previous sections have introduced some subtype relationships involving
+extension types. This section adds further relationships where extension
+types occur more indirectly.
+
+*For instance, earlier sections stated that every extension type is a
+subtype of the top types (like `Object?`), and that `implements` introduces
+a subtype relationship for extension types. Moreover:*
+
+Assume that `E` denotes an extension type that takes `k` type arguments.
+Corresponding to the subtype rule about covariance for classes, a rule is
+introduced which makes `E<S1 .. Sk>` a subtype of `E<T1 .. Tk>` if
+`Sj` is a subtype of `Tj` for each `j` in 1 .. k.
+
+*For example, `E<int>` is a subtype of `E<Object>` because `int` is a
+subtype of `Object`, also in the case where `E` is an extension type.*
+
+The type function `futureValueType` is updated as follows:
+
+- `futureValueType(S?) = futureValueType(S)`, for any `S`.
+- `futureValueType(Future<S>) = S`, for any `S`.
+- `futureValueType(FutureOr<S>) = S`, for any `S`.
+- `futureValueType(void) = void`.
+- `futureValueType(dynamic) = dynamic`.
+- `futureValueType(E) = S` if `E` is an extension type
+  that implements `Future<S>`.
+- Otherwise, `futureValueType(S) = Object?`, for any `S`.
+
+The Dart 1 algorithm which is used to compute the standard upper bound of
+two distinct interface types will work in the same way as it does today,
+albeit on a different superinterface graph:
+
+We introduce an interface type `Any` which declares `operator ==`,
+`hashCode`, `runtimeType`, `noSuchMethod`, `toString` with the same
+signatures as `Object` does today, and `Object` as well as `Null` have
+`Any` as a direct superinterface.
```

Comment:
Let's not get ahead of ourselves. I'm not sure adding `Any` is worth the complexity.

I'd prefer to say that the Dart 1 LUB algorithm is extended to work on super-interface-sets of types which include extension types, and which may not alwaus contain `Object`.
That means that it is possible to have two sets of super-interfaces, whose intersection does not contain any level with exactly one element. 
In that case the LUB algorithm does not have a solution, and the UP step which defers to it then makes the result be `Object?`.

Adding a new type _just_ to avoid the LUB algorithm having zero solutions isn't really worth it.
I'm happy to discuss the value of adding `Any` in general, as a separte feature, but it's not worth it just for this.

Reply:
> I'd prefer to say [something more complex]

I'd prefer the simple and consistent approach, which is to add `Any`. Let's get some feedback from the rest of the language team.